### DISCUSSION TITLE
Relation between `renewable_capacity_factor` and `capacity_factor` for `wind_ppl`

### DISCUSSION THREAD

**Discussion by SongminYu** [iiasa/message_ix#850] [Q&A] [answered]
Hi, I am trying to develop the power system part in a MESSAGE model following this [tutorial](https://github.com/iiasa/message_ix/blob/v3.8.0/tutorial/westeros/westeros_renewable_resource.ipynb). When adding the `wind_ppl`, I noticed that there are two "capacity factor" parameters: `renewable_capacity_factor` and `capacity_factor`. I am a bit confused about their relationship, and how they correspond to the "capacity_factor" in other power system models, i.e., annual full-capacity generation hours divided by 8760.

In the [tutorial](https://github.com/iiasa/message_ix/blob/v3.8.0/tutorial/westeros/westeros_renewable_resource.ipynb), I extracted the values of the two parameters as follows:

(1) `renewable_capacity_factor`

<img width="656" alt="Screenshot 2024-06-05 at 23 48 04" src="https://github.com/iiasa/message_ix/assets/16057131/89cd5360-50a6-46de-b4c9-56a40e66676f">

(2) `capacity_factor`

<img width="665" alt="Screenshot 2024-06-05 at 23 54 23" src="https://github.com/iiasa/message_ix/assets/16057131/6d987209-72d4-4944-afb6-60af20ad8c56">

**Here are my questions:** 

First, what is the relationship between the two parameters? 

Second, I want to know if I understand right: the multiplication of the two "capacity_factor" is comparable with the "capacity_factor" in other power system models, i.e., annual full-capacity generation hours divided by 8760?

Thank you!

**Comment by SongminYu** [comment]
I found some explanation on this [page](https://docs.messageix.org/en/stable/model/MESSAGE/model_core.html#equation-renewables-capacity-requirement) regarding `renewable_capacity_factor`:

> Equation RENEWABLES_CAPACITY_REQUIREMENT
>
> This constraint connects the capacity factor of a renewable grade to the installed capacity of a technology. **_It sets the lower limit for the capacity of a renewable technology to the summed activity over all grades (REN) devided by the capactiy factor of this grade._** It represents the fact that different renewable grades require different installed capacities to provide their full potential.

So, `renewable_capacity_factor` is a factor on the `renewable_potential`, not related to the operational aspect? This **_sentence_** is still not clear to me... 

Besides, next to the `capacity_factor` of technologies, another parameter called `operation_factor` is also mentioned in this [page](https://docs.messageix.org/en/stable/model/MESSAGE/model_core.html#equation-renewables-capacity-requirement). But it is not mentioned in the tutorials. A brief explanation of 

 - `operation_factor` of technologies
 - `capacity_factor` of technologies
 - `renewable_capacity_factor` of renewable resources

would be much appreciated!

Thanks!

**Comment by glatterf42** [comment, marked as answer]
Hi @SongminYu and sorry for the late reply! 
In general, I think your understanding is correct, but let me try to add a little more context here. Maybe @SiddharthJoshi-Git can jump in and correct me if I'm wrong :pray: 

So when you define renewable energy technologies, they have the usual `capacity_factor` that all technologies have. In contrast with e.g. coal power plants, they do not have a limited supply of resources, though. However, they are also limited by other factors: for wind, there are winds of different velocity, which generate different amounts of energy (I imagine). For the sun, the angle at which the light rays reach us influences how much energy can be captured, among other things. So when dealing with renewable energy technologies, we have to capture this inefficiency with another factor: `renewable_capacity_factor`. 
So assume you have a renewable technology with only one `grade`, then the model will use the renewable technologies only to the limit that `ACT/CAP` needs to be smaller or equal to the product of `capacity_factor`, `renewable_capacity_factor`, and `operation_factor`. In practice, the `operation_factor` is often set to 1, and people call the product of `capacity_factor` and `renewable_capacity_factor` the `true_capacity_factor`. 
The `operation_factor` is defined as the ratio of the number of hours the technology was on-line (i.e. active) to the total number of hours in the reference period. So this allows you to define that a technology needs to be inactive every so-and-so hours for e.g. maintenance. 
Of course, when you begin to introduce multiple `grade`s of a renewable technology, the calculation is not as straightforward to arrive at something like a `true_capacity_factor`, but the general idea remains the same: `capacity_factor` describes the fraction of installed capacity that can be active during any one period, while `renewable_capacity_factor` describes the fraction of this renewable energy grade's potential that can be used. 

Please let me know if this answers your question :)

**Reply by SongminYu** [reply]
Hi @glatterf42, thanks for your explanation! It is clear to me now.
